### PR TITLE
Improve Eady figure

### LINKED ISSUE #140: Figure 1 from JOSS paper
This should be an easy fix.

As mentioned in https://github.com/openjournals/joss-reviews/issues/5669#issuecomment-1666968906

The axis tick labels in this figure is unnecessarily small. If you cannot find a way to make it larger, you can remove it and mention the dimensions of the domain in the caption.

All other fonts from the images are also small, but not as small as these tick labels. 
I usually try to use fonts around the same size as the main text.

image

### PR DESCRIPTION
**Pull request by jagoosw**
This PR addresses #140 and increases the model resolution.

Closes #140

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ An example of a problem involving small-scale flow features is showcased in \aut
...
 To this physical setup, we added a medium complexity (9 tracers) biogeochemical model, some components of which are shown above.
 On top of this, we added particles modelling the growth of sugar kelp, which are free-floating and advected by the flow, and carbon dioxide exchange from the air.
-Thanks to Julia's speed and efficiency the above model (1 km × 1 km × 100 m with 64 × 64 × 16 grid points) took about 30 minutes of computing time to simulate 10 days of evolution on an Nvidia P100 GPU. 
+Thanks to Julia's speed and efficiency the above model (1 km × 1 km × 100 m with 512 × 512 × 64 grid points) took about X minutes of computing time to simulate 10 days of evolution on an Nvidia A100 GPU. 
 Panel (a) shows the domain with the colour representing the concentration of various biogeochemical tracer fields: inorganic carbon, organic carbon (dissolved and particulate), phytoplankton, and nutrients. 
 The increase in organic carbon concentration in the centre of the eddy can be seen, as well as carbon being subducted (most visible in the xz face in the organic carbon).
```
Comment (jagoosw) [inline paper/paper.md:77]: @johnryantaylor I will update this caption when I have made the new figure, but do you want to add a sentence about the physics visible at the high resolution?

File: paper/paper.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ An example of a problem involving small-scale flow features is showcased in \aut
 ![In this simulation of baroclinic instability in the Eady problem, a background buoyancy gradient and corresponding thermal wind generates a sub-mesoscale eddy, roughly following the setup of @taylor:2016.
```
Comment (johnryantaylor) [inline paper/paper.md:72]:
```suggestion
![In this simulation of baroclinic instability in the Eady problem, a background buoyancy gradient and corresponding thermal wind generates a sub-mesoscale eddy, roughly following the setup of @taylor:2016. A submesoscale front develops on the periphery of the eddy where intense three-dimensional turbulence develops with inherently non-hydrostatic dynamics.
```

File: paper/paper.md (hunk 1 above)
Comment (johnryantaylor) [reply paper/paper.md:77]: Good point - I suggested a sentence above

File: paper/figures/eady.jl [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                        sim.model.clock.iteration,
+                        prettytime(sim.model.clock.time),
+                        prettytime(sim.run_wall_time),
+                        prettytime(sim.Δt))
+
+simulation.callbacks[:progress] = Callback(progress, IterationInterval(10))
+
+u, v, w = model.velocities
+
+# Periodically save the velocities and vorticity to a file.
```
Comment (navidcy) [inline paper/figures/eady.jl:119]:
```suggestion
# Periodically save the velocities to a file.
```

File: paper/figures/eady.jl [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+B_field = BackgroundField(B, parameters = background_state_parameters)
+
+νᵥ = κᵥ = 1e-4 # [m² s⁻¹]
+vertical_diffusivity = VerticalScalarDiffusivity(ν = νᵥ, κ = κᵥ)
+
+@info "Setting up kelp particles"
+
+n = 32
+
+x = arch_array(arch, [repeat([-200.0, -100.0, 0.0, 100.0, 200.0], 1, 5)...] .+ Lx / 2)
```
Comment (navidcy) [inline paper/figures/eady.jl:55]:
```suggestion
x = arch_array(arch, [repeat([-200.0, -100.0, 0.0, 100.0, 200.0], 1, 5)...]  .+ Lx / 2)
```

File: paper/figures/eady.jl [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+y = arch_array(arch, [repeat([-200.0, -100.0, 0.0, 100.0, 200.0], 1, 5)'...] .+ Ly / 2)
+z = arch_array(arch, zeros(Float64, n))
+
+particles = SLatissima(; architecture = arch,
+                         x, y, z, 
+                         A = arch_array(arch, 5.0 .* ones(n)), N = arch_array(arch, 0.01.* ones(n)), C = arch_array(arch, 0.18.* ones(n)),
```
Comment (navidcy) [inline paper/figures/eady.jl:61]:
```suggestion
                         A = arch_array(arch, 5.0 .* ones(n)), N = arch_array(arch, 0.01 .* ones(n)), C = arch_array(arch, 0.18 .* ones(n)), 
```

File: paper/figures/eady.jl [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+# Periodically save the velocities and vorticity to a file.
+simulation.output_writers[:fields] = JLD2OutputWriter(model, merge(model.tracers, (; u, v, w));
+                                                      schedule = TimeInterval(6hours),
+                                                      filename = "eady.jld2",
+                                                      overwrite_existing = true)
+
+simulation.output_writers[:particles] = JLD2OutputWriter(model, (; particles);
+                                                          schedule = TimeInterval(6hours),
+                                                          filename = "eady_particles.jld2",
+                                                          overwrite_existing = true)
```
Comment (navidcy) [inline paper/figures/eady.jl:128]:
```suggestion
                                                         schedule = TimeInterval(6hours),
                                                         filename = "eady_particles.jld2",
                                                         overwrite_existing = true)
```

File: paper/figures/eady.jl [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+n = 32
+
+x = arch_array(arch, [repeat([-200.0, -100.0, 0.0, 100.0, 200.0], 1, 5)...] .+ Lx / 2)
+y = arch_array(arch, [repeat([-200.0, -100.0, 0.0, 100.0, 200.0], 1, 5)'...] .+ Ly / 2)
+z = arch_array(arch, zeros(Float64, n))
+
+particles = SLatissima(; architecture = arch,
+                         x, y, z, 
+                         A = arch_array(arch, 5.0 .* ones(n)), N = arch_array(arch, 0.01.* ones(n)), C = arch_array(arch, 0.18.* ones(n)), 
+                         latitude = 57.5,
```
Comment (navidcy) [inline paper/figures/eady.jl:62]: where does this latitude value comes from? is this the value that corresponds to the `f` value we use above?

Comment (jagoosw) [reply paper/figures/eady.jl:62]: It was arbitrary for fast kelp growth, but I've changed now to correspond to the coriolis force.

File: paper/paper.md [hunk 8]
```diff
@@ -72,7 +72,7 @@ An example of a problem involving small-scale flow features is showcased in \aut
...
 To this physical setup, we added a medium complexity (9 tracers) biogeochemical model, some components of which are shown above.
 On top of this, we added particles modelling the growth of sugar kelp, which are free-floating and advected by the flow, and carbon dioxide exchange from the air.
-Thanks to Julia's speed and efficiency the above model (1 km × 1 km × 100 m with 64 × 64 × 16 grid points) took about 30 minutes of computing time to simulate 10 days of evolution on an Nvidia P100 GPU. 
+Thanks to Julia's speed and efficiency the above model (1 km × 1 km × 100 m with 512 × 512 × 64 grid points) took about X minutes of computing time to simulate 10 days of evolution on an Nvidia A100 GPU.
```
[the remark this answers is not in the document]
Comment (jagoosw) [reply paper/paper.md:75]: Yeah not quite finished running!

File: paper/figures/eady.jl (hunk 4 above)
Comment (jagoosw) [reply paper/figures/eady.jl:55]: I'll update this since there was an issue here too

**Comment by jagoosw** [comment]
Do we think the eady plot is now sufficiently clear and large enough text?

**Comment by jagoosw** [comment]
It doesn't have the colorbars or kelp line plot but I don't think they add much to the demonstration value of the plot.

**Comment by jagoosw** [comment]
I think I would like to add to one of the sentences about performance an acknowledgement of and reference for KA/JuliaGPU

**Comment by jagoosw** [comment]
I've also just noticed that the reference for Oceananigans isn't at the first instance it was mentioned, should we change?

**Comment by navidcy** [comment]
> Do we think the eady plot is now sufficiently clear and large enough text?

Fontsizes for axis labels don't seem to have increased. Have they?
Also lack of colorbars makes it less clear for me.
Another suggestion is to alter a bit the colorrange for the left-most part of the figure so that the y-z structure shows up better?

File: src/Boundaries/Sediments/simple_multi_G.jl [hunk 9]
```diff
@@ -170,56 +174,56 @@ sediment_fields(model::SimpleMultiG) = (C_slow = model.fields.C_slow,
...
-
-        Δz = grid.Δzᵃᵃᶜ[1]
-
-        timestepper.Gⁿ.NH₄[i, j, 1] += Nᵐⁱⁿ * (1 - pₙᵢₜ) / Δz
-        timestepper.Gⁿ.NO₃[i, j, 1] += Nᵐⁱⁿ * pₙᵢₜ / Δz
-        timestepper.Gⁿ.DIC[i, j, 1] += Cᵐⁱⁿ / Δz
-        timestepper.Gⁿ.O₂[i, j, 1]  -= max(0, (1 - pᵈᵉⁿⁱᵗ - pₐₙₒₓ * pₛₒₗᵢ) * Cᵐⁱⁿ / Δz) # this seems dodge but this model doesn't cope with anoxia properly
+                     sediment.denitrification_params.E * log(reactivity) ^ 2 +
+                     sediment.denitrification_params.F * log(O₂) * log(reactivity)) / (Cᵐⁱⁿ * day)
+        =#
```
Comment (navidcy) [inline src/Boundaries/Sediments/simple_multi_G.jl:209]: shall we delete these commented code?

File: src/Boundaries/Sediments/simple_multi_G.jl [hunk 10]
```diff
@@ -170,56 +174,56 @@ sediment_fields(model::SimpleMultiG) = (C_slow = model.fields.C_slow,
...
+                    E * log(O₂) * log(reactivity) +
+                    F * log(NO₃) ^ 2) / (Cᵐⁱⁿ * day)
+
+        A, B, C, D = sediment.solid_dep_params
+        pₛₒₗᵢ = A * (C * depth ^ D) ^ B
+
+        tendencies.NH₄[i, j, k] += Nᵐⁱⁿ * (1 - pₙᵢₜ) / Δz
+        tendencies.NO₃[i, j, k] += Nᵐⁱⁿ * pₙᵢₜ / Δz
+        tendencies.DIC[i, j, k] += Cᵐⁱⁿ / Δz
+        tendencies.O₂[i, j, k]  -= max(0, ((1 - pₐₙₒₓ * pₛₒₗᵢ) * Cᵐⁱⁿ + 2 * Nᵐⁱⁿ * pₙᵢₜ)/ Δz) # this seems dodge but this model doesn't cope with anoxia properly (I think)
```
Comment (navidcy) [inline src/Boundaries/Sediments/simple_multi_G.jl:226]:
```suggestion
        tendencies.O₂[i, j, k]  -= max(0, ((1 - pₐₙₒₓ * pₛₒₗᵢ) * Cᵐⁱⁿ + 2 * Nᵐⁱⁿ * pₙᵢₜ)/ Δz) # this seems dodgy but it seems like this model doesn't cope with anoxia properly
```
?

File: src/Boundaries/Sediments/simple_multi_G.jl (hunk 9 above)
Comment (jagoosw) [reply src/Boundaries/Sediments/simple_multi_G.jl:209]: I guess so, might be better todo in #138 since I've made a lot of changes to the sediment there

**Comment by jagoosw** [comment]
> Fontsizes for axis labels don't seem to have increased. Have they? Also lack of colorbars makes it less clear for me. Another suggestion is to alter a bit the colorrange for the left-most part of the figure so that the y-z structure shows up better?

I think the axis labels appear the same size as other figures now? I think the main complaint was about the ticks.

Is this better with color bars:
eady

I've adjusted the colorrange so it clips on the visible faces (rather than the whole volumes min/max), is that okay?

**Comment by navidcy** [comment]
> > Fontsizes for axis labels don't seem to have increased. Have they? Also lack of colorbars makes it less clear for me. Another suggestion is to alter a bit the colorrange for the left-most part of the figure so that the y-z structure shows up better?
> 
> I think the axis labels appear the same size as other figures now? I think the main complaint was about the ticks.

Oh I see! Okie then!

> Is this better with color bars: eady

Yes! Much better!

> I've adjusted the colorrange so it clips on the visible faces (rather than the whole volumes min/max), is that okay?

Try 0.8*min / max ? Ideally I'd like a bit more contrast for the left-part of the figure. But it's also fine as is!

**Comment by jagoosw** [comment]
I think this is a bit better:
eady_example

**Review by navidcy — APPROVED** [review]

**Pull request merged**